RecoLocalTracker/SiPixelRecHits/plugins/PixelRecHits.cu

```diff
@@ -59,27 +90,51 @@ namespace pixelgpudetails {
       input.nDigis,
       gpu_.hitsModuleStart_d,
       gpu_.charge_d,
-      gpu_.xg_d, gpu_.yg_d, gpu_.zg_d,
-      gpu_.xerr_d, gpu_.yerr_d, gpu_.mr_d,
-      true // for the time being stay local...
+      gpu_.detInd_d,
+      gpu_.xg_d, gpu_.yg_d, gpu_.zg_d, gpu_.rg_d,
+      gpu_.iphi_d,
+      gpu_.xl_d, gpu_.yl_d,
+      gpu_.xerr_d, gpu_.yerr_d, 
+      gpu_.mr_d, gpu_.mc_d
     );
 
     // needed only if hits on CPU are required...
     cudaCheck(cudaMemcpyAsync(hitsModuleStart_, gpu_.hitsModuleStart_d, (gpuClustering::MaxNumModules+1) * sizeof(uint32_t), cudaMemcpyDefault, stream.id()));
+   
+    // to be moved to gpu?
+    auto nhits = hitsModuleStart_[gpuClustering::MaxNumModules];
+    uint32_t hitsLayerStart[11];
+    for (int i=0;i<10;++i) hitsLayerStart[i]=hitsModuleStart_[phase1PixelTopology::layerStart[i]];
+    hitsLayerStart[10]=nhits;
+
+    std::cout << "hit layerStart "; 
+    for (int i=0;i<10;++i) std::cout << phase1PixelTopology::layerName[i] << ':' << hitsLayerStart[i] << ' ';
+    std::cout << "end:" << hitsLayerStart[10] << std::endl;
+
+    cudaCheck(cudaMemcpyAsync(gpu_.hitsLayerStart_d, hitsLayerStart, (11) * sizeof(uint32_t), cudaMemcpyDefault, stream.id()));
```

Comment:
I believe this is not safe because `hitsLayerStart` is a local array. It should have a life time longer than the async copy takes (effectively to be a member of `PixelRecHitGPUKernel`).